app/models/user.rb

```diff
@@ -12,10 +12,31 @@ class User < ApplicationRecord
   devise :database_authenticatable,
          :recoverable, :rememberable, :validatable
 
-
   has_many :cart_items, dependent: :delete_all
   has_many :works_in_cart, through: :cart_items, source: :work
 
+  # This will correspond to a "role" in the AccessPolicy class.
+  # "editor" will replace the current "staff" role.
+  # A new "reader" type will be added in a future PR.
+  USER_TYPES = %w{admin editor}.freeze
+  enum user_type: USER_TYPES.collect {|v| [v, v]}.to_h
+  validates :user_type, presence: true
+
+
+  def admin?
+    raise RuntimeError.new("Deprecated. Use admin_user?")
```

Comment:
if you are actually raising, that's not just a deprecation -- and it will never ever get to the next line 28, the exception will stop it!

A deprecation would just maybe print a warning but actually work...

But maybe there's no need to have a deprecation phase in our own app, just remove the method? What do you think?

Reply:
I actually want the error (happy to not call it a "deprecation"). I definitely want to remove the method later on in the process -- after we drop the old `admin` column though.

Reply:
I'm not following why we want the error. 

With this error, if any code calls `admin?`, it'll get a `RuntimeError.new("Deprecated. Use admin_user?")`.   

Without this error, if any code calls `admin?`, it'll just get a `NoMethodError` I think?  Or will something else happen?  

Is it just that you prefer the former error becuase you or I might be confused by `NoMethodError`? Thoughts:

1. Nothing in the code should be calling `admin?` if it does it's a bug -- so this method should actually never be called.  Do we really want to leave a method in that should never be called?

I would personally remove it, becuase we control this code completely, there should simply be no code that still calls `admin?` anywhere, if we want to remove the method. If there accidentally is, the `NoMethodError` is good enough, compared to a custom error.  But no big deal. 

3. If it is to remain, "deprecated" is the wrong word -- something that is deprecated still works but will disappear in the future. This one as here does not still work. Also there is no reason to leave line 28 there, which is impossible to ever be reached. If the method is just going to raise, that can be the entirety of the method body.

Reply:
Ah right, because the `admin` column means Rails does provide an `admin?` ethod, and we want to turn that into an error!

good call!

OK fine. just don't call it "deprecated" and remove the rest of the method body! :)